docs/introduction/how-to-work-with-money.md

```diff
@@ -0,0 +1,388 @@
+
+It works similarly to `priceWithCurrency`, but it uses currency ID instead of the `Currency` entity.
+
+The *currency ID* (`int`) must be provided as a parameter.
+
+### priceWithCurrencyByDomainId
+
+It works similarly to `priceWithCurrency`, but it uses the basic frontend currency of the provided domain.
+
+The *domain ID* (`int`) must be provided as a parameter.
+
+## Money in Javascript
+
+`Money` implements the `JsonSerializable` interface, which means it can be serialized into JSON format with `json_encode` into an object with the following structure:
+
+```json
+{
+  "amount": "1234.5670"
+}
+```
+
+This format can be readily used in Javascript when JSON is rendered in Twig:
+
+```twig
+{# the "money" variable contains Money::create('1234.5670') #}
+
+<div id="my-money" data-money="{{ money|json_encode }}"/>
+```
+
+```js
+(function ($) {
+
+    $(document).ready(function () {
+        var money = $('#my-money').data('money');
+
+        // Displays "1234.5670"
+        alert(money.amount);
+    });
+
+})(jQuery);
+```
+
+See the native Javascript method [`.toFixed()`](https://www.w3schools.com/jsref/jsref_tofixed.asp) and the function [`parseFloat()`](https://www.w3schools.com/jsref/jsref_parsefloat.asp) to see how to convert between decimal strings and numbers.
+
+## Money in Doctrine
+
+In Shopsys Framework, there is a custom Doctrine type `money` which can be used in similar fashion as `decimal` type.
+The entity property value will be automatically hydrated to an instance of `Money` if it's configured to use the type:
+
+```php
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="my_entities")
+ * @ORM\Entity
+ */
+class MyEntity
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Money\Money|null
+     *
+     * @ORM\Column(type="money", precision=20, scale=6, nullable=true)
+     */
+    protected $price;
+
+    // ...
+}
+```
+
+When you want to use a `Money` instance as a parameter in DQL, use the `getAmount` method in your [repository class](/docs/introduction/basics-about-model-architecture.md#repository):
+
+```php
+use Doctrine\ORM\Query\Expr\Join;
+use Shopsys\FrameworkBundle\Component\EntityExtension\QueryBuilder;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductCalculatedPrice;
+
+class MyProductFilterRepository
+{
+    // ...
+
+    /**
+     * @param \Doctrine\ORM\QueryBuilder $builder
+     * @param \Shopsys\FrameworkBundle\Component\Money\Money $limit
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup $group
+     */
+    protected function filterByMinimalPrice(QueryBuilder $builder, Money $limit, PricingGroup $group)
+    {
+        $condition = 'pcp.product = p AND pcp.pricingGroup = :group AND pcp.priceWithVat >= :limit';
+
+        $productsQueryBuilder->join(ProductCalculatedPrice::class, 'pcp', Join::WITH, $condition)
```

Comment:
the example is too long. Please use only the `filterByMinimalPrice` method and remove `$groupParameter`.
You can maybe add link to actual usage for correctness

Reply:
I've already severely shortened the example. Having it shorter may hurt actual understandability in my opinion. I'm not sure in which context it might work without `PricingGroup`.

Would this be better?

```php
use Shopsys\FrameworkBundle\Component\Money\Money;

// ...

$priceLimit = Money::create(1000);

/** @var \Doctrine\ORM\QueryBuilder $builder */
$builder->setParameter('priceLimit', $priceLimit->getAmount());
```

Reply:
much better :+1:

Reply:
Replaced in 1b497aec1